src/Modules/OrchardCore.Commerce/Services/FieldsOnlyDisplayManager.cs

```diff
@@ -0,0 +1,107 @@
+using Lombiq.HelpfulLibraries.OrchardCore.Contents;
+using Microsoft.AspNetCore.Http;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata;
+using OrchardCore.ContentManagement.Metadata.Models;
+using OrchardCore.ContentManagement.Metadata.Settings;
+using OrchardCore.DisplayManagement;
+using OrchardCore.Templates.Controllers;
+using OrchardCore.Templates.Services;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Commerce.Services;
+
+public class FieldsOnlyDisplayManager : IFieldsOnlyDisplayManager
+{
+    private readonly IContentDefinitionManager _contentDefinitionManager;
+    private readonly IHttpContextAccessor _hca;
+    private readonly IShapeFactory _shapeFactory;
+    private readonly TemplatesManager _templatesManager;
+
+    public FieldsOnlyDisplayManager(
+        IContentDefinitionManager contentDefinitionManager,
+        IHttpContextAccessor hca,
+        IShapeFactory shapeFactory,
+        TemplatesManager templatesManager)
+    {
+        _contentDefinitionManager = contentDefinitionManager;
+        _hca = hca;
+        _shapeFactory = shapeFactory;
+        _templatesManager = templatesManager;
+    }
+
+    public IEnumerable<string> GetFieldShapeTypes(
+        ContentItem contentItem,
+        string displayType = CommonContentDisplayTypes.Detail)
+    {
+        var typeDefinition = _contentDefinitionManager.GetTypeDefinition(contentItem.ContentType);
+        return typeDefinition
+            .Parts
+            .SelectMany(part =>
+                part.PartDefinition.Fields.Select(field => new
+                {
+                    PartName = part.Name,
+                    FieldName = field.Name,
+                    PartOrder = GetNumericOrder(part),
+                    FieldOrder = GetNumericOrder(field),
+                }))
+            .OrderBy(item => item.PartOrder)
+            .ThenBy(item => item.FieldOrder)
+            .Select(item => $"{contentItem.ContentType}_{displayType}__{item.PartName}__{item.FieldName}");
+    }
+
+    public async Task<IEnumerable<IShape>> DisplayFieldsAsync(
+        ContentItem contentItem,
+        string displayType = CommonContentDisplayTypes.Detail)
+    {
+        var fieldShapeTypes = GetFieldShapeTypes(contentItem, displayType);
+
+        var shapes = new List<IShape>();
+        foreach (var shapeType in fieldShapeTypes)
+        {
+            shapes.Add(await _shapeFactory.CreateAsync(shapeType));
+        }
```

Comment:
Isn't something like this would simpler? 
```suggestion
        var shapes = (await Task.WhenAll(fieldShapeTypes.Select(async shapeType =>
            await _shapeFactory.CreateAsync(shapeType)))).ToList();
```